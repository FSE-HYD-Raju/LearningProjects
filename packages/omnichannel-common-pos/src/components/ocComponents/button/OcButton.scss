@import '../../../styles/_variables.scss';

.OcButton {
    display: flex;
    justify-content: center;
    align-items: center;

    &:hover {
        >a {
            text-decoration: none;
        }
    }

    >a {
        text-decoration: none;
        width: 100%;
    }

    >span {
        text-decoration: none;
        width: 100%;
        &:hover {
            text-decoration: none;
        }
    }

    &-error {
		//@include ocbutton();
		color: theme-color("danger");
		border: 1px solid theme-color("danger");
		background: white;
		&:hover {

		}
    }

    &-error_filled {
		//@include ocbutton();
        color: white;
		border: 1px solid theme-color("danger");
        background: theme-color("danger");
        color: white;
        &:hover {
            background: lighten(theme-color("danger"), 5%);
            color: white;
        }
    }

	&-large {
		font-size: $font-size-base;
		max-height: 40px;
        min-height: 40px;
	}

	&-ghost {
		//@include ocbutton();
		//border: 1px solid #DCDCDC;
        border: 1px solid white;
		color: theme-color("primary");
        background: white;
        box-shadow: 0 0 4px #e0e0e0;
		&:hover {
			border: 1px solid #DCDCDC;
			color: lighten(#555, 5%);
		}
	}

	&-ghost_filled {

	}

	&-borderless {
		border: 1px solid transparent;
		background: transparent;
		color: theme-color("primary");

		&:hover {
			color: lighten(theme-color("primary"), 10%);
		    border: 1px solid transparent;
		}
	}

	&-small {
		max-height: 29px !important;
		padding: 4px 8px !important;
	}

	&-block {
		width: 100%;
	}

	&-disabled {
		opacity: 0.3;
        cursor: not-allowed;
        pointer-events: none;
	}
}
