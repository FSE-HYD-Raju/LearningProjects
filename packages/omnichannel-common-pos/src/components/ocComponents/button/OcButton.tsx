import * as React from "react";
import * as classnames from "classnames";

enum OcButtonType {
	SECONDARY = "SECONDARY",
	DEFAULT = "DEFAULT",
	PRIMARY = "PRIMARY",
	SUCCESS = "SUCCESS",
	DANGER = "DANGER",
	LINK = "LINK",
}

enum OcButtonSize {
	SMALL = "SMALL",
	LARGE = "LARGE",
	DEFAULT = "DEFAULT",
}

enum OcButtonDropdownType {
	SPLIT = "SPLIT",
	REGULAR = "REGULAR",
}

type OcButtonHtmlButtonType = "submit" | "reset" | "button";

interface OcButtonProps extends React.ClassAttributes<HTMLButtonElement> {
	children?: React.ReactNode;
	onClick?: (event: React.SyntheticEvent<HTMLButtonElement>) => void;
	buttonType?: keyof typeof OcButtonType;
	buttonSize?: keyof typeof OcButtonSize;
	htmlBtnType?: OcButtonHtmlButtonType;
	className?: string;
	outline?: boolean;
	disabled?: boolean;
	id?: string;
	icon?: string;
	block?: boolean;
	dropdownType?: keyof typeof OcButtonDropdownType;
	style?: React.CSSProperties;
	title?: string;
}

const OcButtonInternal: React.RefForwardingComponent<HTMLButtonElement, OcButtonProps> = (props: OcButtonProps, ref: React.Ref<HTMLButtonElement>) => {
	const { buttonType, outline, buttonSize, block, disabled, className = "", dropdownType, icon, id, htmlBtnType = "button" } = props;
	const buttonClasses = classnames({
		btn: true,
		"btn-outline": !buttonType && outline,
		"btn-secondary": buttonType && (!outline && buttonType === OcButtonType.SECONDARY),
		"btn-outline-secondary": buttonType && (outline && buttonType === OcButtonType.SECONDARY),
		"btn-primary": buttonType && (!outline && buttonType === OcButtonType.PRIMARY),
		"btn-outline-primary": buttonType && (outline && buttonType === OcButtonType.PRIMARY),
		"btn-success": buttonType && (!outline && buttonType === OcButtonType.SUCCESS),
		"btn-outline-success": buttonType && (outline && buttonType === OcButtonType.SUCCESS),
		"btn-danger": buttonType && (!outline && buttonType === OcButtonType.DANGER),
		"btn-outline-danger": buttonType && (outline && buttonType === OcButtonType.DANGER),
		"btn-link": buttonType && buttonType === OcButtonType.LINK,
		"btn-lg": buttonSize === OcButtonSize.LARGE,
		"btn-sm": buttonSize === OcButtonSize.SMALL,
		"dropdown-toggle": dropdownType,
		"btn-block": block,
		"btn-icon-link": icon,
		disabled: disabled,
		[className]: Boolean(className)
	});
	return (
		<button
			type={htmlBtnType}
			className={buttonClasses}
			disabled={disabled}
			onClick={props.onClick}
			id={id}
			style={props.style}
			title={props.title}
			key={props.key}
			ref={ref}
		>
			{icon && <i className={`fa fa-fw ${icon}`}/> }
			{icon && " "}
			{dropdownType && dropdownType === OcButtonDropdownType.SPLIT && <span className="caret"/> }
			{dropdownType && dropdownType === OcButtonDropdownType.REGULAR &&
				<span>
					{props.children}{" "}<span className="caret"/>
				</span>}
			{!dropdownType && props.children}
		</button>
	);
};
OcButtonInternal.displayName = "OcButton";

const OcButton: React.ForwardRefExoticComponent<OcButtonProps & React.RefAttributes<HTMLButtonElement>> =
	React.forwardRef<HTMLButtonElement, OcButtonProps>(OcButtonInternal);
OcButton.displayName = "OcButton";
export {
	OcButtonDropdownType,
	OcButtonSize,
	OcButtonType,
	OcButtonProps,
	OcButtonHtmlButtonType,
	OcButton,
};
