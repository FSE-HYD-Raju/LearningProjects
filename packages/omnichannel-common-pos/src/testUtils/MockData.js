import _ from "lodash";
import R from "ramda";

const firstDateThisWeek = () => {
	const date = new Date();
	while (date.getDay() !== 0) {
		date.setDate(date.getDate() - 1);
	}
	return date;
};

const people = [
	{
		firstName: "Kathrin",
		lastName: "K",
		gender: "female",
		language: "swe",
		countryOfBirth: "SE",
		birthDay: firstDateThisWeek(),
		emails: [{ email: "kathrin.k@qvantel.com" }],
		email: "kathrin.k@qvantel.com",
		mobileNumbers: [{ number: "2398457" }],
		mobileNumber: "2398457",
		fixedLineNumbers: [{ number: "2398457" }],
		fixedLineNumber: "2398457",
		password: "qwerasdf",
		passwordConfirmation: "qwerasdf"
	},
	{
		firstName: "John",
		lastName: "Smith",
		gender: "male",
		language: "fin",
		birthDay: firstDateThisWeek(),
		countryOfBirth: "United Kingdom",
		emails: [{ email: "john.smith@qvantel.com" }],
		email: "john.smith@qvantel.com",
		mobileNumbers: [{ number: "3498573" }],
		mobileNumber: "3498573",
		fixedLineNumbers: [{ number: "52497584" }],
		fixedLineNumber: "52497584"
	}
];

const basketItems = [
	{
		id: "7879c247-9e98-40ee-bb48-544eb366494a",
		type: "basketItems",
		attributes: {
			product: {
				name: "Movie Steaming Service"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 1.9,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 14.9,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: [],
			totalUpfrontPrice: 1.9,
			totalUpfrontPriceWithChildren: 1.9
		}
	},
	{
		id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
		type: "basketItems",
		attributes: {
			product: {
				name: "XDSL"
			},
			totalPrices: [
				{
					type: "ONE_TIME",
					taxFreeAmount: 100.6,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: 30,
					taxRate: 0,
					currency: "EUR"
				}
			],
			childBasketItems: []
		}
	}
];

const deliveryItem = {
	id: "c0065ffc-77f4-4e44-b15c-fbd2f78d04ce",
	type: "basketItems",
	inputtedCharacteristics: {
		CH_Delivery_Contact_Info: "5c5dddf6-72be-4d0a-9711-6027f247af28"
	},
	product: {
		name: "Home delivery",
		id: "PO_Home_Delivery"
	}
};

export default class MockData {
	static getPerson(searchCriteria) {
		if (searchCriteria) {
			return _.find(people, searchCriteria);
		} else {
			return people[0];
		}
	}

	static getPeople() {
		return people;
	}

	static getPostalAddress() {
		return {
			street: "1 Test St",
			coAddress: "Best Ltd.",
			postalCode: "90210",
			city: "Test City",
			country: "FI"
		};
	}

	static getUser() {
		return {
			"id": "bf588945-ce95-490a-a774-8ef91661a146",
			"type": "persons",
			"attributes": {
				"lastName": "Testeri",
				"addresses": null,
				"gender": "male",
				"fixedLineNumbers": [],
				"formattedName": "Teppo Testeri",
				"roles": null,
				"billingAccountId": null,
				"language": null,
				"emails": [
					{
						"role": "PRIMARY",
						"id": "2ab79d7b-862d-49df-8afc-e6104a914919",
						"email": "tepsteri@tepinposti.fi"
					}
				],
				"honorificSuffix": null,
				"countryOfBirth": "FI",
				"personRelationships": [],
				"birthDay": null,
				"placeOfBirth": null,
				"characteristics": null,
				"mobileNumbers": [
					{
						"role": "PRIMARY",
						"id": "a843f46d-f729-40dd-a7ce-240a64bc008e",
						"number": "1234576890"
					}
				],
				"honorificPrefix": null,
				"avatar": "/img/icon-user-male.svg",
				"privacySettings": {
					"privacy1": false,
					"privacy2": false
				},
				"postalAddresses": [
					{
						"role": "PRIMARY",
						"id": "5e30221d-ad19-4043-8088-f594f5fdf77f",
						"street": "Piippukatu 11 A 1",
						"coAddress": null,
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": null,
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					},
					{
						"role": "DELIVERY",
						"id": "5c5dddf6-72be-4d0a-9711-6027f247af28",
						"street": "Piippukatu 11 A 1",
						"coAddress": "Piippukatu 11 A 1",
						"postalCode": "40100",
						"city": "Roma",
						"country": "FI",
						"county": null,
						"description": null,
						"postOfficeBox": null,
						"stateOrProvince": "",
						"addressRegisterId": null,
						"apartment": null,
						"building": null
					}
				],
				"identifications": [],
				"firstName": "Teppo",
				"customerAccountId": null,
				"nationality": null,
				"additionalName": null,
				"maritalStatus": null
			}
		}
	}

	static getBasket(withItems: boolean, withDeliveryItems: boolean) {
		const basket = {
			activeBasket: {
				attributes: {
					totalPrices: [],
					totalUpfrontPrice: 0
				}
			}
		};

		if (withItems) {
			basket.basketItems = basketItems;

			const totalPriceOneTime = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);
			const totalPriceMonthly = R.reduce(
				R.add,
				0,
				basketItems.map(bi => {
					return bi.attributes.totalPrices[0].taxFreeAmount;
				})
			);

			basket.activeBasket.attributes.totalPrices = [
				{
					type: "ONE_TIME",
					taxFreeAmount: totalPriceOneTime,
					taxRate: 0,
					currency: "EUR"
				},
				{
					type: "RECURRENT",
					taxFreeAmount: totalPriceMonthly,
					taxRate: 0,
					currency: "EUR"
				}
			];

			if (withDeliveryItems) {
				basket.basketItems[0].attributes.childBasketItems.push(
					deliveryItem
				);
			}
		}

		return basket;
	}
}
