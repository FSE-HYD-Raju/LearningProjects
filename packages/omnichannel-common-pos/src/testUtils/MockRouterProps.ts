import {
	RouteComponentProps,

} from "react-router";

import {
	Href,
	UnregisterCallback,
	TransitionPromptHook,
	LocationState,
	Path,
	LocationDescriptorObject,
	LocationListener
} from "history";

const mockLocation: RouteComponentProps<any>["location"] = {
		hash: "",
		key: "",
		pathname: "",
		search: "",
		state: {}
};

const mockHistory: RouteComponentProps<any>["history"] = {
	length: 2,
	action: "POP",
	location: mockLocation,
	push: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	replace: (path: Path | LocationDescriptorObject, state?: LocationState): void => {},
	go: (n: number): void => {},
	goBack: (): void => {},
	goForward: (): void => {},
	block: (prompt?: boolean | string | TransitionPromptHook): UnregisterCallback => { return () => {}; },
	listen: (listener: LocationListener): UnregisterCallback => {return () => {}; },
	createHref: (location: LocationDescriptorObject): Href => { return ""; },
};

const mockRouterProps: RouteComponentProps<any> = {
		match: {
			isExact: true,
			params: {},
			path: "",
			url: ""
		},
		location: mockLocation,
		history: mockHistory,
		staticContext: {}
};

export {
	mockRouterProps,
	mockLocation,
	mockHistory,
	RouteComponentProps
};
