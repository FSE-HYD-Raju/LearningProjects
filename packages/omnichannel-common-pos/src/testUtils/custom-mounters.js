import React from "react";
import { IntlProvider, intlShape } from "react-intl";
import { mount, shallow } from "enzyme";
import PropTypes from "prop-types";
import R from "ramda";
import TestUtils from "./TestUtils";

const MAX_DIVE_DEEP_LEVEL = 16;
const intlProvider = new IntlProvider({ locale: "en" }, {});
const { intl } = intlProvider.getChildContext();

const baseContext = {
	flux: {
		persist: () => {}
	},
	store: TestUtils.mockReduxStore(),
	intl,
	location: {},
	only: {},
	skip: {}
};

const nodeWithIntlProp = (node, mockedRouterOptions) => {
	if (mockedRouterOptions) {
		mockedRouterOptions.props().history.push = jest.fn();
	}
	return React.cloneElement(node, {
		intl,
		...(mockedRouterOptions ? mockedRouterOptions.props() : {})
	});
};

function getOptionsForShallowMount(options, debug, mockedRouterContextOptions) {
	const mountOptions = R.merge(options ? options : {}, {
		disableLifecycleMethods: !(
			options && options.disableLifecycleMethods === false
		),
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				intl: intlShape,
				store: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mountOptions;
}
function deepDive(wrapper) {
	let diveAttemptsCount = 0;
	while (
		wrapper.hostNodes().length === 0 &&
		diveAttemptsCount < MAX_DIVE_DEEP_LEVEL
	) {
		wrapper = wrapper.dive();
		diveAttemptsCount += 1;
	}
	return wrapper;
}
export function shallowWithContextAndRouterProps(node, options, debug) {
	return shallowWithContext(node, options, debug, true);
}
export function shallowWithContextAndLifecycle(node, options, debug) {
	return shallowWithContext(
		node,
		R.merge(options || {}, { disableLifecycleMethods: false }),
		debug,
		true
	);
}
export function deepShallowWithContextAndLifecycle(node, options, debug) {
	return deepDive(shallowWithContextAndLifecycle(node, options, debug));
}
export function deepShallowWithContext(node, options, debug) {
	return deepDive(shallowWithContext(node, options, debug, true));
}
export function shallowWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	return shallow(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		getOptionsForShallowMount(options, debug, mockedRouterContextOptions)
	);
}

export function mountWithContextAndRouterProps(node, options, debug) {
	return mountWithContext(node, options, debug, true);
}
export function mountWithContext(node, options, debug, injectRouterProps) {
	const mockedRouterOptions = TestUtils.getMockedRouterOptions();
	const mockedRouterContextOptions = mockedRouterOptions.get();
	const mountOptions = R.merge(options ? options : {}, {
		context: R.mergeAll([
			debug
				? {
						...baseContext
					}
				: baseContext,
			R.omit("intl", options && options.context ? options.context : {}),
			mockedRouterContextOptions.context
		]),
		childContextTypes: R.merge(
			{
				flux: PropTypes.object,
				store: PropTypes.object,
				intl: intlShape,
				location: PropTypes.object,
				only: PropTypes.object,
				skip: PropTypes.object
			},
			mockedRouterContextOptions.childContextTypes
		)
	});
	TestUtils.mockReduxStoreToFlux(mountOptions.context);
	return mount(
		nodeWithIntlProp(node, injectRouterProps ? mockedRouterOptions : null),
		mountOptions
	);
}

const attachmentElementId = "__OC_COMPONENT_TEST_ATTACHMENT_ELEMENT__";

export function attachWithContext(node, options) {
	const attachmentElement = document.createElement("div");
	attachmentElement.setAttribute("id", attachmentElementId);
	document.body.appendChild(attachmentElement);

	const wrapper = mountWithContext(
		node,
		R.merge(options, { attachTo: attachmentElement })
	);
	const doppelWrapper = hijackDetacher(wrapper);

	return doppelWrapper;
}

function hijackDetacher(wrapper) {
	const doppel = wrapper;
	doppel.__detach = wrapper.detach;

	doppel.detach = () => {
		removeAttachmentElements();
		doppel.__detach.apply(wrapper);
	};

	return doppel;
}

function removeAttachmentElements() {
	document.body.querySelectorAll("#" + attachmentElementId).forEach(elem => {
		document.body.removeChild(elem);
	});
}

export function detach(wrapper) {
	removeAttachmentElements();
	wrapper.detach();
}
