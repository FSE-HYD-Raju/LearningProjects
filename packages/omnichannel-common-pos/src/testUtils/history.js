import isClient from "../utils/isClient";
import { createBrowserHistory, createMemoryHistory } from "history";
import { useRouterHistory } from "react-router";

const history = isClient
	? useRouterHistory(createBrowserHistory)({ queryKey: false })
	: createMemoryHistory();

export default history;
