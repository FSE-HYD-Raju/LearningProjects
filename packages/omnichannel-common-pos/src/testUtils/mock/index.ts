import { default as productOffering } from "./ProductOfferingMock";
import { ProductMock as product } from "./ProductMock";
import { AgreementMock as agreement } from "./AgreementMock";
import { ChargingBalancesMock as chargingBalances } from "./ChargingBalancesMock";
import { default as postalAddress } from "./PostalAddressMock";
import { default as identification } from "./IdentificationMock";
import { default as user } from "./UserMock";
import { default as order } from "./OrderMock";
import { default as basket } from "./BasketMock";
export { productOffering, postalAddress, identification, user, order, basket, product, agreement, chargingBalances };
