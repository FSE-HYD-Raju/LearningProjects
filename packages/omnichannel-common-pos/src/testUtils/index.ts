import ReactWidgetsTestUtils from "./ReactWidgetsTestUtils";
import * as MockDataMaker from "./mock";
const {
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach
} = require("./custom-mounters");
const TestUtils = require("./TestUtils");
const MockData = require("./MockData");
const SimpleDataMock = require("./mock/simpleDataMock");

export {
	MockData,
	ReactWidgetsTestUtils,
	TestUtils,
	SimpleDataMock,
	shallowWithContextAndRouterProps,
	shallowWithContextAndLifecycle,
	deepShallowWithContextAndLifecycle,
	deepShallowWithContext,
	shallowWithContext,
	mountWithContextAndRouterProps,
	mountWithContext,
	attachWithContext,
	detach,
	MockDataMaker
};

export * from "./MockRouterProps";

export default TestUtils;
