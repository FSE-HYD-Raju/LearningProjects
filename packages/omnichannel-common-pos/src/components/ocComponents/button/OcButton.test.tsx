import * as React from "react";
import { mountWithContext, shallowWithContext } from "../../../testUtils";
import { OcButton, OcButtonProps, OcButtonSize, OcButtonType } from "./OcButton";
import { ReactWrapper } from "enzyme";

const getAllCssClasses = (wrapper: ReactWrapper<OcButtonProps>): Array<string> => {
	const allClasses: Array<string> = [];
	wrapper.getDOMNode().classList.forEach((className: string) => {
		allClasses.push(className);
	});
	return allClasses.sort();
};

describe("OcButton", () => {
	it("succeeds at shallow mount with minimum props", () => {
		const wrapper = shallowWithContext(<OcButton />);
		expect(wrapper).toMatchSnapshot();
	});

	it("succeeds at deep mount with minimum props", () => {
		mountWithContext(<OcButton />);
	});

	it("adds only btn css class to html button when no props provided", () => {
		const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton />);
		expect(getAllCssClasses(wrapper)).toEqual(["btn"]);
	});

	it("adds btn and btn-outline css classes to html button for outline property active", () => {
		const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton outline={true} />);
		expect(getAllCssClasses(wrapper)).toEqual(["btn", "btn-outline"]);
	});

	it("adds btn and btn-block css classes to html button for block property active", () => {
		const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton block={true} />);
		expect(getAllCssClasses(wrapper)).toEqual(["btn", "btn-block"]);
	});

	it("sets type=\"button\" to html button by default", () => {
		const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton />);
		expect(wrapper.getDOMNode().getAttribute("type")).toEqual("button");
	});

	it("sets appropriate button type when htmlBtnType property is set", () => {
		const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton htmlBtnType="submit" />);
		expect(wrapper.getDOMNode().getAttribute("type")).toEqual("submit");
		wrapper.setProps({htmlBtnType: "reset"});
		wrapper.update();
		expect(wrapper.getDOMNode().getAttribute("type")).toEqual("reset");
		wrapper.setProps({htmlBtnType: undefined});
		wrapper.update();
		expect(wrapper.getDOMNode().getAttribute("type")).toEqual("button");
	});

	it("sets appropriate css class when buttonSize property is set", () => {
		const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton />);
		expect(getAllCssClasses(wrapper).includes("btn-sm")).toEqual(false);
		wrapper.setProps({buttonSize: OcButtonSize.SMALL});
		wrapper.update();
		expect(getAllCssClasses(wrapper).includes("btn-sm")).toEqual(true);

		wrapper.setProps({buttonSize: OcButtonSize.LARGE});
		wrapper.update();
		expect(getAllCssClasses(wrapper).includes("btn-lg")).toEqual(true);
	});

	it("calls onClick handler when it is set", () => {
		const onClickFn = jest.fn();
		const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton onClick={onClickFn} />);
		wrapper.simulate("click");
		expect(onClickFn).toHaveBeenCalled();
		expect(onClickFn).toHaveBeenCalledTimes(1);
	});

	it("adds custom className when it is provided", () => {
		const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton className="custom-button" />);
		expect(getAllCssClasses(wrapper).includes("custom-button")).toEqual(true);
	});

	it("adds disabled css class when dispabed property is active", () => {
		const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton disabled={true} />);
		expect(getAllCssClasses(wrapper).includes("disabled")).toEqual(true);
	});

	describe("check buttonType", () => {
		Object.keys(OcButtonType).forEach((buttonType: string) => {

			it(`adds btn-${buttonType.toLowerCase()} classes to html button for ${buttonType} button type`, () => {
				const wrapper: ReactWrapper<OcButtonProps> = mountWithContext(<OcButton
					buttonType={buttonType as keyof typeof OcButtonType}/>);
				let buttonTypeClass = `btn-${buttonType.toLowerCase()}`;
				expect(wrapper.hasClass("btn"));

				if (buttonType !== OcButtonType.DEFAULT) {
					expect(wrapper.hasClass(buttonTypeClass));
					expect(getAllCssClasses(wrapper)).toEqual(["btn", buttonTypeClass]);

					wrapper.setProps({outline: true});
					wrapper.update();

					// link does not supports outline type
					if (wrapper.prop("buttonType") !== OcButtonType.LINK) {
						buttonTypeClass = `btn-outline-${buttonType.toLowerCase()}`;
					}
					expect(wrapper.hasClass(buttonTypeClass));
					expect(getAllCssClasses(wrapper)).toEqual(["btn", buttonTypeClass]);
				}
			});
		});
	});
});
