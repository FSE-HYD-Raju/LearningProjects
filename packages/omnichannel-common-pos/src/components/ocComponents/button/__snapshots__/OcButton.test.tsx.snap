// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcButton succeeds at shallow mount with minimum props 1`] = `
<button
  className="btn"
  type="button"
/>
`;
